OoyalaSkinSampleApp/app/src/main/java/com/ooyala/sample/players/SsaiPlayerActivity.java

```diff
@@ -0,0 +1,84 @@
+package com.ooyala.sample.players;
+
+import android.os.Bundle;
+import android.util.Log;
+
+import com.ooyala.android.OoyalaPlayer;
+import com.ooyala.android.PlayerDomain;
+import com.ooyala.android.analytics.IqConfiguration;
+import com.ooyala.android.configuration.Options;
+import com.ooyala.android.skin.OoyalaSkinLayout;
+import com.ooyala.android.skin.OoyalaSkinLayoutController;
+import com.ooyala.android.skin.configuration.SkinOptions;
+import com.ooyala.android.ssai.OoyalaSsaiManager;
+import com.ooyala.sample.R;
+
+import org.json.JSONObject;
+
+
+public class SsaiPlayerActivity extends AbstractHookActivity {
+    private OoyalaSsaiManager ssaiManager;
+    private String playerParams = "";
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Bundle extras = getIntent().getExtras();
+        if (extras != null) {
+            playerParams = extras.getString("player_params");
+        }
+        setContentView(R.layout.player_skin_simple_layout);
+        completePlayerSetup(asked);
+    }
+
+    @Override
+    void completePlayerSetup(boolean asked) {
```

Comment:
What does 'asked' mean here?

Reply:
On the `AbstractHookActivity` class, we have logic that will prompt the user for permissions to store info on the SD card. The pop up appears before the player is shown or created, after the popup is closed the `asked` parameter becomes true and that's when we create the player. That logic is shared amongst all the player Activities in this sample app.